### PR TITLE
#41 publish script only for master branch

### PR DESCRIPTION
**Pull request by sergeevNick**

### CODE REVIEW

File: src/scripts/project/bundle-dts/bundler.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        walkThrough(
+            fromDir,
+            (dir: string, file: string) => {
+                const fillPath = path.join(dir, file);
+                if (fs.statSync(fillPath).isFile() && /\.d\.ts$/.test(fillPath)) {
+                    files.push(fillPath);
+                }
+            }
+        );
```
Comment (derlagg) [inline src/scripts/project/bundle-dts/bundler.ts:18]: Would be better for reading to move callback to method

File: src/scripts/project/update-deps/dependency-worker.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        parsedPackageJson[dependencyType] = { ...parsedPackageJson[dependencyType], ...latestDependenciesMap };
+
+        const indent = 4;
+        const packageJsonStringified = JSON.stringify(parsedPackageJson, undefined, indent);
+        fs.writeFileSync('package.json', packageJsonStringified);
+    }
+
+    private run(command: string): void {
+        // eslint-disable-next-line no-console
+        console.log('run command:', command);
```
Comment (derlagg) [inline src/scripts/project/update-deps/dependency-worker.ts:119]: Let's use Logger

File: src/scripts/project/update-deps/dependency-worker.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    private restoreLatestDependencies(dependencyType: string, latestDependencies: string[]): void {
+        if (!latestDependencies.length) {
+            return;
+        }
+
+        const parsedPackageJson = JSON.parse(fs.readFileSync('package.json', 'utf8'));
+        const latestDependenciesMap = latestDependencies.reduce((acc, curr) => {
+            return { ...acc, [curr]: 'latest' };
+        }, {} as Dictionary<string>);
```
Comment (derlagg) [inline src/scripts/project/update-deps/dependency-worker.ts:109]: Anton is always saying about usage reduce method to get dictionary. He would be upset :(

File: src/scripts/project/update-deps/dependency-worker.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        if (!isFullUpdate) {
+            return packages.filter(dependency => dependencyStore[dependency]);
+        }
+
+        const snapshotPattern = /-/;
+        return Object.keys(dependencyStore)
+            .filter(key => {
+                const isSnapshot = snapshotPattern.test(dependencyStore[key]);
+                const isUrl = dependencyStore[key].startsWith('http');
+                const isLatest = dependencyStore[key] === 'latest';
```
Comment (derlagg) [inline src/scripts/project/update-deps/dependency-worker.ts:81]: This check looks like duplicate for findLatestDependencies function. It's bad

File: package.json [hunk 5]
```diff
@@ -30,7 +30,7 @@
         "test:ci": "jest --ci",
         "hooks:pre-push": "yarn lint && yarn test",
         "hooks:pre-commit": "lint-staged",
-        "deploy:publish": "node ./dist/scripts/ci/publish.js",
+        "deploy:publish": "node ./dist/scripts/ci/publish",
```
[the remark this answers is not in the document]
Comment (sergeevNick) [reply package.json:33]: Better to clarify that we use index.js

File: package.json [hunk 6]
```diff
@@ -46,22 +46,23 @@
...
     },
     "dependencies": {
         "@eigenspace/argument-parser": "1.0.2"
     },
     "devDependencies": {
-        "@eigenspace/codestyle": "1.1.60",
-        "@eigenspace/common-types": "0.0.21-dev",
+        "@eigenspace/codestyle": "3.0.3",
+        "@eigenspace/commit-linter": "0.0.21",
+        "@eigenspace/common-types": "0.0.20",
```
[the remark this answers is not in the document]
Comment (sergeevNick) [reply package.json:57]: Update to 21

File: src/scripts/ci/publish/index.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { Publisher } from './publisher';
+import { ArgumentParser } from '@eigenspace/argument-parser';
+
+const argParser = new ArgumentParser();
+const argv = argParser.get(process.argv.slice(2));
+
+const branch = argv.get('branch') as string;
```
[the remark this answers is not in the document]
Comment (sergeevNick) [reply src/scripts/ci/publish/index.ts:7]: We can do checkout inside script if we provide branch

File: src/scripts/project/update-deps/dependency-worker.ts (hunk 2 above)
Comment (sergeevNick) [reply src/scripts/project/update-deps/dependency-worker.ts:119]: Set console log avoid restriction of whole file

**Pull request merged**